File: LeetCode/2718. Sum of Matrix After Queries/solution.py
class Solution:
    def matrixSumQueries(self, n: int, queries: List[List[int]]) -> int:
        ans = 0
        rows = [0]*n 
        cols = [0]*n 
        cntr = cntc = 0
        for t,idx,v in queries[::-1]:
            if t == 0:
                if rows[idx] == 0:
                    rows[idx] = 1
                    cntr += 1
                    ans += (n-cntc)*v 
            else:
                if cols[idx] == 0:
                    cols[idx] = 1
                    cntc += 1
                    ans += (n-cntr)*v 
        return ans 

File: LeetCode/2718. Sum of Matrix After Queries/solution.cpp
class Solution {
public:
    long long matrixSumQueries(int n, vector<vector<int>>& queries) {
        vector<unordered_set<int>> visited{unordered_set<int>(),unordered_set<int>()};
        long long ans = 0;
        int q = queries.size();
        for (int i = q - 1; i >= 0; i--) {
            int tpe = queries[i][0];
            int idx = queries[i][1];
            long long val = queries[i][2];
            if (visited[tpe].count(idx)) continue;
            visited[tpe].insert(idx);
            ans += val*(n-(int)visited[1-tpe].size());
        }
        return ans;
    }
};